src/mkdir.js

```diff
@@ -28,8 +28,14 @@ function mkdirSyncRecursive(dir) {
   // Base dir does not exist, go recursive
   mkdirSyncRecursive(baseDir);
 
-  // Base dir created, can create dir
-  fs.mkdirSync(dir, parseInt('0777', 8));
+  try {
+    // Base dir created, can create dir
+    fs.mkdirSync(dir, parseInt('0777', 8));
```

Comment:
It looks like this same bug can happen above on line 24 (the other call to `fs.mkdirSync`). What do you think about refactoring this to recurse first (if necessary) and only call `fs.mkdirSync` in one spot?

```js
// Replace lines 22-39 with...

// Base dir does not exist, recursion necessary
if (!fs.existsSync(baseDir)) {
  mkdirSyncRecursive(baseDir);
}

try {
  // Base dir created, can create dir
  // etc. ...
}
```

Reply:
Sure, I'll push a change as a "fixup! ..." commmit.
If it's corrent, tests ok, and you like it, I'll squash it into the initial commit prior to a merge.

Reply:
Tests are passing, commits are squashed.